build/v1/types.go

```diff
@@ -85,7 +85,7 @@ type CommonSpec struct {
 	// If set to an empty map or a map with any values, default build nodeselector values
 	// are ignored.
 	// +optional
-	NodeSelector OptionalNodeSelector `json:"nodeSelector" protobuf:"bytes,9,name=nodeSelector"`
+	NodeSelector OptionalNodeSelector `json:"nodeSelector,omitempty" protobuf:"bytes,9,name=nodeSelector"`
```

Comment:
Unfortunately we can't use omit empty here because of forward compatibilty (nil and empty map mean two different things).  See the godoc.

Reply:
Unless we overrode the behavior of OptionalNodeSelector JSON unmarshal (something you can check)

Reply:
No we don't have a custom marshaller for it. I've reverted that part of the pull